ums-core/src/main/java/org/ums/persistent/dao/PersistentApplicationTESDao.java

```diff
@@ -0,0 +1,749 @@
+          + "          b.FIRST_NAME, "
+          + "          b.LAST_NAME, "
+          + "          MST_DEPT_OFFICE.SHORT_NAME, "
+          + "          a.DEPT_OFFICE, "
+          + "          c.DESIGNATION_NAME "
+          + "          from EMPLOYEES a,EMP_PERSONAL_INFO b,MST_DESIGNATION c,MST_DEPT_OFFICE,TES_COURSE_ASSIGN d WHERE a.DEPT_OFFICE=?  AND a.EMPLOYEE_TYPE=1 AND "
+          + "          a.EMPLOYEE_ID=b.EMPLOYEE_ID AND a.DESIGNATION=c.DESIGNATION_ID AND "
+          + "          MST_DEPT_OFFICE.DEPT_ID=a.DEPT_OFFICE AND d.TEACHER_ID=a.EMPLOYEE_ID AND d.SEMESTER_ID=?";
+
+  String getSemesterParameter =
+      "SELECT SEMESTER_ID,to_char(START_DATE,'DD-MM-YYYY') START_DATE,to_char(END_DATE,'DD-MM-YYYY') END_DATE from MST_PARAMETER_SETTING WHERE SEMESTER_ID=? and PARAMETER_ID=?";
+
+  String getFacultyListForReport =
+      "select DISTINCT TEACHER_ID from TES_COURSE_ASSIGN WHERE  DEPT_ID=? AND SEMESTER_ID=?";
+
+  String getAllFacultyListForReport = "select DISTINCT TEACHER_ID from TES_COURSE_ASSIGN WHERE SEMESTER_ID=?";
+
+  String getParameterForReport =
+      "SELECT DISTINCT COURSE_ID,TEACHER_ID,SEMESTER_ID,DEPT_ID from TES_COURSE_ASSIGN WHERE  TEACHER_ID=? AND SEMESTER_ID=?";
+  String getDeptList = "SELECT DEPT_ID from MST_DEPT_OFFICE WHERE TYPE=1 ORDER  BY DEPT_ID";
+
+  String getTotalRegisteredStudentForCourse =
+      "SELECT  COUNT (a.STUDENT_ID) FROM UG_REGISTRATION_RESULT_CURR a,STUDENTS b WHERE  a.COURSE_ID=? AND b.THEORY_SECTION=? AND a.SEMESTER_ID=? "
+          + "AND  a.STUDENT_ID=b.STUDENT_ID AND b.DEPT_ID=(select DEPT_ID from MST_PROGRAM where PROGRAM_ID=( "
+          + "select PROGRAM_ID from SEMESTER_SYLLABUS_MAP where SEMESTER_ID=? and (SYLLABUS_ID, Year, semester) in ( "
+          + "select SYLLABUS_ID, MST_COURSE.\"YEAR\", MST_COURSE.SEMESTER from COURSE_SYLLABUS_MAP, MST_COURSE WHERE COURSE_SYLLABUS_MAP.COURSE_ID= ? and "
+          + "MST_COURSE.COURSE_ID=COURSE_SYLLABUS_MAP.COURSE_ID)))";
+
+  String getQuestions =
+      "SELECT QUESTION_ID,QUESTION_DETAILS,SEMESTER_ID,OBSERVATION_TYPE FROM APPLICATION_TES_QUESTIONS";
+  String getQuestionSemesterMap =
+      "select QUESTION_ID,SEMESTER_ID,STATUS from APPLICATION_TES_SET_QUESTIONS WHERE SEMESTER_ID=?";
+  String getSectionList =
+      "SELECT COURSE_ID,ASSIGNED_SECTION from TES_COURSE_ASSIGN WHERE COURSE_ID=? AND SEMESTER_ID=? AND TEACHER_ID=? ORDER BY ASSIGNED_SECTION";
+  String getAllSectionForSelectedCourse =
+      "select COURSE_ID,\"SECTION\" from COURSE_TEACHER WHERE COURSE_ID=? AND SEMESTER_ID=? AND TEACHER_ID=? ORDER BY \"SECTION\"";
+  String getDeptListByFacultyId = "SELECT DISTINCT  DEPT_ID FROM MST_PROGRAM WHERE CATEGORY_ID=?";
+  String getCourseForQuestionWiseReport =
+      "SELECT DISTINCT a.COURSE_ID FROM MST_COURSE a,COURSE_TEACHER b WHERE a.COURSE_TYPE=1 AND a.OFFER_BY=? AND a.\"YEAR\"=? AND a.SEMESTER=? AND b.SEMESTER_ID=? AND a.COURSE_ID=b.COURSE_ID";
+
+  String getTeacherListForQuestionWiseReport =
```

Comment:
Should it here?

Reply:
Question wise Report e Parameters
1.Dept
2.year
3.Semester 
4.Semester Id
5.Question Id
eta tahole kothai dibo ami?